lib/ProductOpener/Food.pm

```diff
@@ -4053,6 +4061,12 @@ sub special_process_product($) {
 		$product_ref->{pnns_groups_1} = "unknown";
 		$product_ref->{pnns_groups_1_tags} = ["unknown", "missing-association"];
 	}
+
+	# Put back the original categories_tags so that they match what is in the taxonomy field
+	# if there is a mistmatch it can cause tags to be added multiple times (e.g. with imports)
+	if (defined $product_ref->{categories_tags}) {
```

Comment:
Could the tags be `categories_tags` changed to `undefined` in the meanwhile? It _might_ be safer to set `@{$product_ref->{categories_tags}}` if  `@original_categories_tags` has any elements.

Reply:
Good idea, i'll change it.